Challenge-3/src/helper.ts

```diff
@@ -0,0 +1,106 @@
+import { Contract } from "@ethersproject/contracts";
+import { Provider } from "@ethersproject/providers";
+import { Alert, AlertsResponse, Finding } from "forta-agent";
+import {
+  OPT_ESCROW_ADDRESS,
+  DAI_ADDRESS,
+  L2_ABI,
+  ESCROW_ABI,
+  DAI_L2_ADDRESS,
+} from "./constants";
+import { BigNumber } from "ethers";
+import { createFinding } from "./findings";
+import { getAlerts } from "forta-agent";
+
+const L2Alert: Alert = {
+  alertId: "L2-SUPPLY",
+  hasAddress: () => false,
+  metadata: {
+    optEscBal: BigNumber,
+    abtEscBal: BigNumber,
+  },
+};
+
+const l2Alerts: AlertsResponse = {
+  alerts: [L2Alert],
+  pageInfo: { hasNextPage: false },
+};
+
+export default class Helper {
+  private provider: Provider;
+  constructor(provider: Provider) {
+    this.provider = provider;
+  }
+
+  /**
+   * Retrieves the L1 balance for a given address at a specific block number.
+   * @param address - The address for which to retrieve the balance.
+   * @param blockNumber - The block number at which to retrieve the balance.
+   * @returns A Promise that resolves to a string representing the L1 balance.
+   */
+  public async getL1Balance(
+    address: string,
+    blockNumber: number,
+  ): Promise<string> {
+    const L1Contract = new Contract(DAI_ADDRESS, [ESCROW_ABI], this.provider);
+    const balance = await L1Contract.balanceOf(address, {
+      blockTag: blockNumber,
+    });
+
+    const isOptEscrow =
+      address.toLowerCase() === OPT_ESCROW_ADDRESS.toLowerCase();
+    l2Alerts.alerts[0].metadata[isOptEscrow ? "optEscBal" : "abtEscBal"] =
+      balance;
+
+    return balance.toString();
+  }
+
+  public async getL1Alerts(chaindId: number): Promise<AlertsResponse> {
+    return await getAlerts({
+      alertId: "L2-SUPPLY",
```

Comment:
The alert ID is incorrect, as this is trying to fetch the L2 alerts while being on L2.

Reply:
Unresolved the comment above, as the `alertId` is still incorrect.